internal/alertmanager/upstream.go

```diff
@@ -50,10 +40,23 @@ func NewAlertmanager(name, uri string, opts ...Option) error {
 		opt(am)
 	}
 
-	upstreams[name] = am
+	return am
+}
 
-	log.Infof("[%s] Configured Alertmanager source at %s", name, uri)
+// RegisterAlertmanager will add an Alertmanager instance to the list of
+// instances used when pulling alerts from upstreams
+func RegisterAlertmanager(am *Alertmanager) error {
```

Comment:
Would be great if this was on a collection type, rather than modifying a global variable. Again, happy to see it as another change.